src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelCopyRatioEmissionData.java

```diff
@@ -0,0 +1,130 @@
+package org.broadinstitute.hellbender.tools.coveragemodel;
+
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+
+import javax.annotation.Nonnull;
+import java.io.Serializable;
+
+/**
+ * This class stores the data required for calculating the emission probability according to the
+ * GATK Bayesian coverage model:
+ *
+ * <p>
+ *     Mean log multiplicative bias:
+ *     {@link #mu} = m_t + (W.E[z_s])_t
+ * </p>
+ *
+ * <p>
+ *     Sum of target- and sample-specific unexplained variances about the mean:
+ *     {@link #psi} = \Psi_t + \gamma_s
+ * </p>
+ *
+ * <p>
+ *     Raw read count:
+ *     {@link #readCount} = n_{st}
+ * </p>
+ *
+ * <p>
+ *     Additive noise in the Poisson parameter:
+ *     {@link #poissonAdditiveNoise} = [read depth]_s x [mapping error rate]_{st}
+ * </p>
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class CoverageModelCopyRatioEmissionData implements Serializable {
+
+    private static final long serialVersionUID = -7363264674200250712L;
+
+    /**
+     * Mean log multiplicative bias
+     */
+    private final double mu;
+
+    /**
+     * Total unexplained variance
+     */
+    private final double psi;
+
+    /**
+     * Raw read count
+     */
+    private final int readCount;
+
+    /**
+     * Additive noise in the Poisson parameter
+     */
+    private final double poissonAdditiveNoise;
+
+    /**
+     * Sample metadata (sex genotype, depth of coverage, ...)
+     *
+     * @implNote The metadata is necessary for calculating the emission probability, though, they are the same
+     * for all targets. We have separated such global parameters from target-dependent variables and wrapped them
+     * in a class so that they can be "injected" by reference.
+     */
+    private CopyRatioCallingMetadata copyRatioCallingMetadata = null;
+
+    /**
+     * Public constructor.
+     *
+     * @param mu Mean log multiplicative bias
+     * @param psi Total unexplained variance
+     * @param readCount Raw read count
+     * @param poissonAdditiveNoise Additive noise in the Poisson parameter
+     */
+    public CoverageModelCopyRatioEmissionData(final double mu,
+                                              final double psi,
+                                              final int readCount,
+                                              final double poissonAdditiveNoise) {
+        this.mu = ParamUtils.isFinite(mu, "Log multiplicative bias must be finite. Bad value: " + mu);
```

Comment:
@mbabadi The point is not how fast it is to compute the message, because if you're computing the message, the code has failed and an extra few CPU cycles isn't going to make things worse.  The point is the CPU cost when the error message *doesn't* get thrown.  Since Java doesn't have lazy evaluations, `ParamUtils.isFinite(mu, literal + variable)` computes `literal + variable` before evaluating whether `mu` is finite and incurs the cost of string concatenation every time it does the check.  By using a `Supplier` the string computation only occurs if the error is thrown.

The cost of the `Supplier` was negligible compared to concatenation in code I have benchmarked.